pkg/tests/end_to_end_tests/metadata_test.go

```diff
@@ -0,0 +1,202 @@
+// This file and its contents are licensed under the Apache License 2.0.
```

Comment:
these tests need to use the priviliges of prom_reader or prom_writer etc. You should use testhelpers.PgxPoolWithRole function as in the other end_to_end tests